examples/master/master.ino

```diff
@@ -96,48 +99,91 @@ void printHeader(int command, int addr, byte res)
   Serial.print(res);
   Serial.print(" | timeleft : ");
   Serial.print(timeleft);    
-  Serial.println(" ==="); 
+  Serial.println(" ===");
 }
 
 void analyse(int command, int addr, byte res)
 {
-  printHeader(command, addr, res);
+  //printHeader(command, addr, res);
   switch (command)
   {
     case 0: // update timeleft
       break;
     case 1: // defused ?
       if (res == 1)
       {
-        Serial.print("defused : ");
-        Serial.println(addr);
+        //Serial.print("defused : ");
+        //Serial.println(addr);
         addresses[addr] = false;
       }
       break;
     case 2: // difficulty
       break;
     case 3: // user made mistake ?
-      Serial.print("penality : ");
-      Serial.println(res);
-      timeleft = timeleft - res;
+      //Serial.print("penality : ");
+      //Serial.println(res);
+      if (timeleft < res)
+        timeleft = 0;
+      else
+        timeleft = timeleft - res;
       break;
     default:
       break;
   }
 }
 
+bool areAllDefused()
+{
+  for (uint8_t i = 0 ; i < 127 ; ++i)
+  {
+    if (addresses[i] == true)
+      return false;
+  }
+  return true;
+}
+
+void updateLed()
+{
+  if (defused == false)
+  {
+    digitalWrite(RED_LED, HIGH);
+    digitalWrite(GREEN_LED, LOW);
+  }
+  else
+  {
+    digitalWrite(RED_LED, LOW);
+    digitalWrite(GREEN_LED, HIGH);
+  }
+}
+
+void play()
+{
+  if (defused == false)
+  {
+    pingEveryone();
+    if (timeleft != 0)
```

Comment:
timeleft > 0 is more appropriate. For example : in case timeleft goes from 3 to -10, condition is still valid but games continues to run

Reply:
I put a condition to avoid the timeleft to pass under zero and it is an uint16_t (never inferior to 0)

Reply:
On line 125, indeed